tests/test_sync_overrides.sh

```
@@ -0,0 +1,351 @@
+  fi
+fi
+
+# Test 11: override_substitution_for returns 0 + value when key exists.
+sub_override="$WORKDIR/helper-sub.yml"
+cat >"$sub_override" <<'YAML'
+substitutions:
+  phase_4b_default: fallback-only
+YAML
+val=$(override_substitution_for "$sub_override" "phase_4b_default") \
+  && [ "$val" = "fallback-only" ] \
+  && pass "override_substitution_for returns override value" \
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)